src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_multi.h

```diff
@@ -0,0 +1,258 @@
+ * \brief Pushes all of the filtration values of a simplextree onto a grid, c.f. \ref find_coordinates.
+ * \ingroup multiparameter
+ * \param splxptr Simplextree pointer.
+ * \param grid The multiparameter grid. A vector of size `num_parameters`,
+ * whose elements are the elements of the grid for this axis.
+ * \param coordinate_values If set to true the filtration values will be turned into coordinates in this grid
+ * instead of points in this grid.
+ * */
+void squeeze_filtration(uintptr_t splxptr, const multi_filtration_grid &grid, bool coordinate_values = true) {
+  Simplex_tree<options_multi> &st_multi = *(Gudhi::Simplex_tree<options_multi> *)(splxptr);
+  auto num_parameters = static_cast<unsigned int>(st_multi.get_number_of_parameters());
+  if (grid.size() != num_parameters) {
+    std::cerr << "Bad grid !" << std::endl;
+    throw;
+  }
+  for (const auto &simplex_handle : st_multi.complex_simplex_range()) {
+    auto &simplex_filtration = st_multi.filtration_mutable(simplex_handle);
+    find_coordinates(simplex_filtration,grid);  // turns the simplexfiltration into coords in the grid
+    if (!coordinate_values) {
+      for (auto parameter = 0u; parameter < num_parameters; parameter++)
+        simplex_filtration[parameter] = grid[parameter][simplex_filtration[parameter]];
+    }
+  }
+  return;
+}
+
+// retrieves the filtration values of a simplextree. Useful to generate a grid.
+/**
+ * \brief Retrieves all of the filtration values, for each simplex dimension, of the simplextree. Useful to generate
+ * grids. \ingroup multiparameter \param splxptr Simplextree pointer. whose elements are the elements of the grid for
+ * this axis. \param degrees Only the simpleces of these dimension will be taken into account. Useful for, e.g., Rips
+ * filtrations.
+ * */
+std::vector<multi_filtration_grid> get_filtration_values(const uintptr_t splxptr, const std::vector<int> &degrees) {
+  Simplex_tree<options_multi> &st_multi = *(Gudhi::Simplex_tree<options_multi> *)(splxptr);
+  int num_parameters = st_multi.get_number_of_parameters();
+  std::vector<multi_filtration_grid> out(degrees.size(), multi_filtration_grid(num_parameters));
+  std::vector<int> degree_index(degrees.size());
+  int count = 0;
+  for (auto degree : degrees) {
+    degree_index[degree] = count++;
+    out[degree_index[degree]].reserve(st_multi.num_simplices());
+  }
+
+  for (const auto &simplex_handle : st_multi.complex_simplex_range()) {
+    const auto filtration = st_multi.filtration(simplex_handle);
+    const auto degree = st_multi.dimension(simplex_handle);
+    if (std::find(degrees.begin(), degrees.end(), degree) == degrees.end()) continue;
+    for (int parameter = 0; parameter < num_parameters; parameter++) {
+      out[degree_index[degree]][parameter].push_back(filtration[parameter]);
+    }
+  }
+  return out;
+}
+
+}  // namespace Gudhi::multiparameter
+
+namespace std {
+
+template <>
+class numeric_limits<Gudhi::multiparameter::multi_filtration_type> {
```

Comment:
I am not sure this code is doing what it is suppose to do:
```cpp
  using Multi_Filtration_values = multi_filtrations::Finitely_critical_multi_filtration<float>;
  std::clog << std::numeric_limits<Multi_Filtration_values>::quiet_NaN() << std::endl;
  // displays: [ ]
  std::clog << std::numeric_limits<Multi_Filtration_values>::infinity() << std::endl;
  // displays: [ ]
```

Reply:
I changed the position of this part of the code to the Finitely critical filtration file, which is more logical, and retested on my laptop and I get, as expected `[nan]` and `[inf]`. Does this fix works for you ?

Reply:
Yes !